network/src/main/scala/com/linkedin/norbert/network/NodeSpecification.scala

```diff
@@ -0,0 +1,118 @@
+/**
+ * Partitioned and non-partitioned NodeSpecification wrapper objects for sendRequest
+ */
+package com.linkedin.norbert.network
+
+import com.linkedin.norbert.network.javaobjects.{NodeSpecification => JNodeSpecification, PartitionedNodeSpecification => JPartitionedNodeSpecification}
+
+/**
+ * A NodeSpecification object is used to store the necessary information to specify a node.
+ * For the non-partitioned version this is the capability and persistentCapability.
+ * The Partitioned version extends the non-partitioned version with numberOfReplicas and clusterId
+ */
+
+/**
+ * NodeTrait is the trait that the NodeSpecification objects extend.
+ * @tparam NodeType Either a non-partitioned or partitioned nodeSpec
+ */
+trait NodeTrait[NodeType] {
+  var capability: Option[Long] = None
+  var persistentCapability: Option[Long] = None
+
+  def setCapability(cap: Option[Long]): this.type = {
```

Comment:
I am little bit skeptical to use this.type as return type definition. Let's make return type more explicit if there is no strong objection.